account_move_transactionid_import/tests/test_completion_transaction_id.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from datetime import datetime
+from odoo.modules import get_resource_path
+from odoo.tests import SingleTransactionCase
+from odoo.tools import convert_file
+
+
+class TestCompletionTransactionId(SingleTransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        convert_file(
+            cls.cr,
+            'account',
+            get_resource_path('account', 'test', 'account_minimal_test.xml'),
+            {},
+            'init',
+            False,
+            'test',
+        )
+        cls.partner = cls.env.ref('base.res_partner_2')
+        cls.journal = cls.env.ref('account.bank_journal')
+        cls.journal.used_for_completion = True
+        cls.move = cls.env['account.move'].create(
+            {'name': 'Move with transaction ID', 'journal_id': cls.journal.id}
+        )
+        cls.move_line = cls.env['account.move.line'].create(
+            {
+                'name': 'Test autocompletion on invoice with transaction ID',
+                'account_id': cls.env.ref('account.a_sale').id,
+                'move_id': cls.move.id,
+                'transaction_ref': 'XXX66Z',
+                'date_maturity': '{}-01-06'.format(datetime.now().year),
+                'credit': 0.0,
+            }
+        )
+
+    def test_sale_order_transaction_id(self):
+        self.move_line.transaction_ref = 'XXX66Z'
+        self.journal.rule_ids = [
```

Comment:
this can be reduced to 
       
       rule_ids': [
                (4, comp_rule.id, False) for comp_rule in rule_ids
            ],